atomix/cluster/src/main/java/io/atomix/raft/roles/FollowerRole.java

```diff
@@ -84,20 +85,11 @@ public FollowerRole(final RaftContext context) {
   public CompletableFuture<InstallResponse> onInstall(final InstallRequest request) {
     final CompletableFuture<InstallResponse> future = super.onInstall(request);
     if (isRequestFromCurrentLeader(request.currentTerm(), request.leader())) {
-      resetHeartbeatTimeout();
+      onHeartbeatFromLeader();
```

Comment:
small question - when the `super.onInstall` takes too long it could happen that we already reach the timeout right?

Reply:
Isn't that an issue?

Reply:
When we cancel the timer any scheduled task related to it will be cancelled too. So even if the timer was triggered during when the install request is being processed, the task won't be executed.